### PR TITLE
Feature/sl 975/optimistick lock

### PR DESCRIPTION
**Pull request by leszekgu**

### CODE REVIEW

File: src/main/resources/db/schema/SL-XXX-add-versioning.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+databaseChangeLog:
+  - changeSet:
+        id: add-versioning
+        author: lolo
```
Comment (radoslawlandowskikainos) [inline src/main/resources/db/schema/SL-XXX-add-versioning.yaml]: Who is lolo? :D

File: src/main/java/uk/gov/hmcts/reform/sandl.snlevents/controllers/SessionController.java [hunk 2]
```diff
@@ -83,10 +86,12 @@ public ResponseEntity insertSession(@RequestBody UpsertSession upsertSession) th
 
     @PutMapping(path = "/update", consumes = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity updateSession(@RequestBody UpsertSession upsertSession) throws IOException {
-
-        UserTransaction ut = sessionService.updateSession(upsertSession);
-
-        return ok(ut);
+        try {
```
Comment (kpakur) [inline src/main/java/uk/gov/hmcts/reform/sandl.snlevents/controllers/SessionController.java]: could not that be handled in a global exception handler so we do not need to do try catch in each controller method that modifies the data

File: src/main/java/uk/gov/hmcts/reform/sandl.snlevents/service/SessionService.java [hunk 3]
```diff
@@ -179,6 +182,7 @@ public UserTransaction saveWithTransaction(UpsertSession upsertSession) {
     public UserTransaction updateSession(UpsertSession upsertSession) throws IOException {
 
         Session session = getSessionById(upsertSession.getId());
+        versionCheck.checkVersion(session, upsertSession.getVersion());
```
Comment (kpakur) [inline src/main/java/uk/gov/hmcts/reform/sandl.snlevents/service/SessionService.java]: not sure what is this for

File: src/main/java/uk/gov/hmcts/reform/sandl.snlevents/service/VersionCheckService.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package uk.gov.hmcts.reform.sandl.snlevents.service;
+
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sandl.snlevents.model.db.VersionedEntity;
+
+import javax.persistence.OptimisticLockException;
+
+@Service
+public class VersionCheckService {
```
Comment (kpakur) [inline src/main/java/uk/gov/hmcts/reform/sandl.snlevents/service/VersionCheckService.java]: consider the entitymanger.merge instead http://blog.xebia.com/jpa-implementation-patterns-saving-detached-entities/

**Pull request merged**